app/src/main/java/org/mozilla/focus/firstrun/UpgradeFirstrunPagerAdapter.java

```diff
@@ -22,6 +24,13 @@ public UpgradeFirstrunPagerAdapter(Context context, View.OnClickListener listene
             this.pages.add(FirstRunLibrary.buildLifeFeedFirstrun(context));
         }
 
-
+        PinSiteManager pinSiteManager = PinSiteManagerKt.getPinSiteManager(context);
+        if (pinSiteManager.isEnabled() && pinSiteManager.isFirstTimeEnable()) {
```

Comment:
1. if the user clear history data,  will he see the onboarding again?
2. Maybe you'll want to use NewFeatureNotice's setLastShownFeatureVersion() to prevent showing this onboarding again. 

Maybe we can land this first, and fix the onboarding issue in another PR?